tests/functional/disaster-recovery/sc_arbiter/test_zone_unaware_app.py

```diff
@@ -0,0 +1,358 @@
+                    continue
+
+                # Fetch the cidrs for creating network fence
+                cidrs = retry(CommandFailed, tries=5)(
+                    get_rbd_daemonset_csi_addons_node_object
+                )(node.name)["status"]["networkFenceClientStatus"][0]["ClientDetails"][
+                    0
+                ][
+                    "cidrs"
+                ]
+
+                # Create the network fence
+                retry(CommandFailed, tries=5)(create_network_fence)(
+                    node.name, cidr=cidrs[0]
+                )
+
+            # Taint the nodes that are shutdown
+            taint_nodes(
+                nodes=[node.name for node in self.nodes_to_shutdown],
+                taint_label=constants.NODE_OUT_OF_SERVICE_TAINT,
+            )
+
+        # Check if all the pods are running
+        log.info(
+            "Checking if all the logwriter/logreader pods are relocated and successfully running"
+        )
+        sc_obj.get_logwriter_reader_pods(label=constants.LOGWRITER_CEPHFS_LABEL)
+        sc_obj.get_logwriter_reader_pods(
+            label=constants.LOGREADER_CEPHFS_LABEL,
+            statuses=[constants.STATUS_RUNNING, constants.STATUS_COMPLETED],
+        )
+        try:
+            retry(UnexpectedBehaviour, tries=1)(sc_obj.get_logwriter_reader_pods)(
+                label=constants.LOGWRITER_RBD_LABEL, exp_num_replicas=2
+            )
+        except UnexpectedBehaviour:
+            if not fencing:
+                log.info(
+                    "It is expected for RBD workload with RWO to stuck in terminating state"
+                )
+                log.info("Trying the workaround now...")
+                pods_terminating = [
+                    Pod(**pod_info)
+                    for pod_info in get_pods_having_label(
+                        label=constants.LOGWRITER_RBD_LABEL,
+                        namespace=constants.STRETCH_CLUSTER_NAMESPACE,
+                    )
+                ]
+                log.info(pods_terminating)
+                for pod in pods_terminating:
+                    log.info(f"Force deleting the pod {pod.name}")
+                    pod.delete(force=True)
+                sc_obj.get_logwriter_reader_pods(
+                    label=constants.LOGWRITER_RBD_LABEL, exp_num_replicas=2
+                )
+            else:
+                log.error(
+                    "Looks like pods are not running or not relocated even after fencing.. please check"
+                )
+                raise
+
+        if fencing:
+
+            # If fencing is True, then unfence the nodes once the pods are relocated
+            log.info(
+                "If fencing was done, then we need to unfence the nodes once the pods are relocated and running"
+            )
+            for node in self.nodes_to_shutdown:
+                if node.name not in worker_nodes:
+                    continue
+                unfence_node(node.name, delete=True)
```

Comment:
can we check fencing was done on node or not before trying for unfence the nodes

Reply:
its been taken care inside unfence_node()